pl/lessons/advanced/escripts.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: page
+title: Uruchamianie programów
+category: advanced
+order: 3
+lang: pl
+---
+
+Do stworzenia pliku wykonywalnego w Elixirze służy escript. Escript generuje plik wykonywalny, który może zostać 
+uruchomiony na każdym komputerze, na którym zainstalowano Erlanga.
+
+{% include toc.html %}
+
+## Na początek
+
+By utworzyć plik, który można uruchomić, za pomocą escriptu musimy zrobić tylko drobnych rzeczy: zaimplementować 
+funkcję `main/1` oraz zaktualizować konfigurację mixa.
+
+Na początek stwórzmy moduł, który będzie punktem startowym programu. W nim zaimplementujemy funkcję `main/1`:
+
+```elixir
+defmodule ExampleApp.CLI do
+  def main(args \\ []) do
+    # Do stuff
+  end
+end
+```
+
+Następnie w pliku mixa dodajemy sekcję `:escript`, która zawiera opcję `:main_module`:
+
+```elixir
+defmodule ExampleApp.Mixfile do
+  def project do
+    [app: :example_app,
+     version: "0.0.1",
+     escript: escript]
+  end
+
+  def escript do
+    [main_module: ExampleApp.CLI]
+  end
+end
+```
+
+## Parsowanie argumentów
+
+Do naszej aplikacji możemy przekazać pewne argumenty z linii poleceń. By je sparsować, użyjemy Elixirowego modułu  
+`OptionParser.parse/2` z opcją `:switches`, która zawiera informacje, iż nasz flaga jest typu logicznego:
+
+```elixir
+defmodule ExampleApp.CLI do
+  def main(args \\ []) do
+    args
+    |> parse_args
+    |> response
+    |> IO.puts
+  end
+
+  defp parse_args(args) do
+    {opts, word, _} =
+      args
+      |> OptionParser.parse(switches: [upcase: :boolean])
+
+    {opts, List.to_string(word)}
+  end
+
+  defp response({opts, "Hello"}), do: response({opts, "World"})
+  defp response({opts, word}) do
+    if opts[:upcase], do: word = String.upcase(word)
+    word
+  end
+end
+```
+
+## Tworzenie plików wykonywalnych
+
+Jak już skonfigurujemy aplikację by używała escript, stworzenie pliku wykonywalnego z mixem jest banalne:
+
+```elixir
+$ mix escript.build
+```
+
+Let's take it for a spin:
+
+```elixir
+$ ./example_app --upcase Hello
+WORLD
+
+$ ./example_app Hi
+Hi
+```
+
```

Comment:
There is lack of: "That’s it. We’ve built our first executable in Elixir using escript." translation at the end.